packages/framework/src/Model/Order/Processing/OrderProcessorMiddleware/ApplyNominalPromoCodeMiddleware.php

```diff
@@ -80,20 +82,32 @@ public function createAndAddOrderItemData(
      * @param \Shopsys\FrameworkBundle\Model\Order\PromoCode\PromoCode $promoCode
      * @param \Shopsys\FrameworkBundle\Model\Order\PromoCode\PromoCodeLimit\PromoCodeLimit $promoCodeLimit
      * @param \Shopsys\FrameworkBundle\Component\Domain\Config\DomainConfig $domainConfig
+     * @param \Shopsys\FrameworkBundle\Model\Order\OrderData $orderData
+     * @param array $validProductIds
      * @return \Shopsys\FrameworkBundle\Model\Order\Item\OrderItemData|null
      */
     protected function createDiscountOrderItemData(
         PromoCode $promoCode,
         PromoCodeLimit $promoCodeLimit,
         DomainConfig $domainConfig,
+        OrderData $orderData,
+        array $validProductIds,
```

Comment:
What about calculating the totalApplicableProductsPrice in the `createAndAddOrderItemData()` method and passing only the calculated value? What do you think?

Reply:
Sure, that seems cleaner.